### PR TITLE
CDAP-1179 Make QueryClient use ClientConfig always, instead of only initially

### PR DESCRIPTION
**Pull request by jwang47**

### CODE REVIEW

File: cdap-explore-client/src/main/java/co/cask/cdap/explore/client/SuppliedAddressExploreClient.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.google.common.base.Supplier;
+
+import java.net.InetSocketAddress;
+
+/**
+ * An Explore Client that uses the supplied host and port to talk to a server
+ * implementing {@link co.cask.cdap.explore.service.Explore} over HTTP.
+ */
+public class SuppliedAddressExploreClient extends AbstractExploreClient {
```
Comment (jguery) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/SuppliedAddressExploreClient.java]: Is it meant to replace ``FixedAddressExploreClient`` ? Does that class not work?

Comment (jwang47) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/SuppliedAddressExploreClient.java]: It works fine, but when the hostname/port is changed via `ClientConfig`, the `FixedAddressExploreClient` doesn't pick up the changes.

Comment (jguery) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/SuppliedAddressExploreClient.java]: I see.

**Comment by jguery** [comment]
LGTM

**Pull request merged**